src/test/scala/com/cognite/sdk/scala/common/RetryWhile.scala

```diff
@@ -17,7 +17,8 @@ trait RetryWhile {
       action: => A,
       assertion: A => Assertion,
       retriesRemaining: Int = 8,
-      initialDelay: FiniteDuration = Constants.DefaultInitialRetryDelay
+      initialDelay: FiniteDuration = Constants.DefaultInitialRetryDelay,
+      retryOnException: Boolean = false
```

Comment:
would probably make sense to accept a predicate of which exceptions to retry, and maybe always retry TestFailedException as before

Reply:
I think the second part (retrying always for `TestFailedException`) is the case now... the first part can probably be added in a separate PR if there's a need to give more granular control over which exceptions to retry